### PR TITLE
Template namespacing.

### PR DESCRIPTION
**Pull request by devel-pa**
I don't have a clue if is better or not, but here is my proposal for namespacing.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -55,8 +57,13 @@ HandlebarsCompiler.prototype.compile = function(data, path, callback) {
     }
 
     if (ns) {
-      key = path.replace(/\\/g,'/').replace(this.pathReplace, '').replace(/\..+?$/, '');
-      result = "if (typeof " + ns + " === 'undefined'){ " + ns + " = {} }; " + ns + "['" + key + "'] = " + source;
+      header = '';
+      if(!this.inited) {
```
Comment (es128) [inline index.js]: Hmm, I think you're relying on the order of compilation matching the order of concatenation, which may not be the case. May be better to try to find a way to add this to the `include` file that ends up in vendor.

Also are you counting on having this defined globally? You'll want to find a way to attach it to an existing object, like something the `handlebars` library exports.

**Comment by devel-pa** [comment]
I thought to attach the function to `Handlebars` object. This is just a proposal.
I think you are talking about `require` file. 

As for the order: if not all the templates are placed in the same file, I agree, can be a problem. I didn't experienced, yet the compilation/concatenation order issue.
And damn, can we have that `beforeCompile` on brunch and plugins? :)))))

**Comment by devel-pa** [comment]
I can't attach to `require`. The configuration is frozen.

**Comment by es128** [comment]
So why not attach to `Handlebars`?

**Comment by devel-pa** [comment]
Attached :)))

**Comment by devel-pa** [comment]
Wow, Thank you

**Pull request merged**